### PR TITLE
[upload - http-extra] fix: download content to file when unsuccessful response (and test) #1750

### LINKED ISSUE #1750: tauri plugin upload doesn't check if an url leads to a 404 when using download()
Instead, the 404 response (e.g. "Not found" or whatever the 404 message is) is downloaded to disk. I don't think this is what should happen. Rather, an error should be thrown there was a 404.

### PR DESCRIPTION
**Pull request by fxsalazar**
fix #1750

### CODE REVIEW

File: plugins/upload/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ serde_json = { workspace = true }
 tauri = { workspace = true }
 log = { workspace = true }
 thiserror = { workspace = true }
-tokio = { version = "1", features = ["fs"] }
+tokio = { version = "1", features = ["fs", "macros"] }
```
Comment (FabianLars) [inline plugins/upload/Cargo.toml]:
```suggestion
tokio = { version = "1", features = ["fs"] }
```
see next suggestion

File: plugins/upload/Cargo.toml [hunk 2]
```diff
@@ -35,3 +35,6 @@ read-progress-stream = "1.0.0"
 native-tls = ["reqwest/native-tls"]
 native-tls-vendored = ["reqwest/native-tls-vendored"]
 rustls-tls = ["reqwest/rustls-tls"]
+
+[dev-dependencies]
+mockito = "1.5.0"
```
Comment (FabianLars) [inline plugins/upload/Cargo.toml:40]:
```suggestion
[dev-dependencies]
mockito = "1.5.0"
tokio = { version = "*", features = ["macros"] }
```

**Review by FabianLars — COMMENTED** [review]
Thank you! Just a tiny code suggestion.

Also, we'd need a changefile (see .changes dir for what it should look like) before we can merge this :)

**Comment by fxsalazar** [comment]
@FabianLars done

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**